ext/ScreenShot/ScreenShot.h

```diff
@@ -5,72 +5,186 @@
 
 #include "../source/Irrlicht/COpenGLBuffer.h"
 #include "../source/Irrlicht/COpenGLExtensionHandler.h"
+#include "irr/asset/IImageAssetHandlerBase.h"
 
 namespace irr
 {
-namespace ext
-{
-namespace ScreenShot
-{
-	
-//! TODO: HANDLE UNPACK ALIGNMENT
-core::smart_refctd_ptr<video::IDriverFence> createScreenShot(video::IDriver* driver, video::IGPUImage* source, video::IGPUBuffer* destination, uint32_t sourceMipLevel=0u, size_t destOffset=0ull, bool implicitflush=true)
-{
-	// will change this, https://github.com/buildaworldnet/IrrlichtBAW/issues/148
-	if (isBlockCompressionFormat(source->getCreationParameters().format))
-		return nullptr;
+	namespace ext
+	{
+		namespace ScreenShot
+		{
+			/*
+				Creates useful FBO that may be used for instance 
+				to fetch default render target color data attachment 
+				after rendering scene.
+				The usage is following:
+				- create frame buffer using the function
+				- blit framebuffers, driver->blitRenderTargets(nullptr, frameBuffer, false, false);
+				Note that in the call above we don't want to copy depth and stencil buffer, but event though default
+				FBO contains depth buffer.
+				- pass frame buffer to performScreenShot(video::IFrameBuffer*)
+				Notes:
+				- color buffer is placed under video::EFAP_COLOR_ATTACHMENT0 attachment
+				- depth buffer is placed under video::EFAP_DEPTH_ATTACHMENT attachment
+			*/
 
-	auto extent = source->getMipSize(sourceMipLevel);
-	video::IGPUImage::SBufferCopy pRegions[1u] = { {destOffset,extent.x,extent.y,{static_cast<asset::IImage::E_ASPECT_FLAGS>(0u),sourceMipLevel,0u,1u},{0u,0u,0u},{extent.x,extent.y,extent.z}} };
-	driver->copyImageToBuffer(source,destination,1u,pRegions);
+			irr::video::IFrameBuffer* createDefaultFBOForScreenshoting(core::smart_refctd_ptr<IrrlichtDevice> device)
+			{
+				auto driver = device->getVideoDriver();
 
-	return driver->placeFence(implicitflush);
-}
+				auto createAttachement = [&](bool colorBuffer)
+				{
+					asset::ICPUImage::SCreationParams imgInfo;
+					imgInfo.format = colorBuffer ? asset::EF_R8G8B8A8_SRGB : asset::EF_D24_UNORM_S8_UINT;
+					imgInfo.type = asset::ICPUImage::ET_2D;
+					imgInfo.extent.width = driver->getCurrentRenderTargetSize().Width;
+					imgInfo.extent.height = driver->getCurrentRenderTargetSize().Height;
```

Comment:
you probably want to use a different function to get the screensize, this will give you the current size of whatever the current FBO bound is